pyroengine/core.py

```diff
@@ -145,17 +142,21 @@ async def analyze_stream(self, image_queue: asyncio.Queue) -> None:
             try:
                 self.engine.predict(frame, cam_id)
             except Exception as e:
-                logging.error(f"Error running prediction: {e}")
+                logger.error(f"Error running prediction: {e}")
             finally:
                 image_queue.task_done()  # Mark the task as done
 
     async def night_mode(self) -> bool:
         """
-        Checks if it is nighttime for any camera.
-
-        Returns:
-            bool: True if it is daytime for all cameras, False otherwise.
+        Checks and updates the day_time attribute based on the current frame.
         """
+        try:
```

Comment:
why do you need to add this ? we are already checking all cameras just below, you will do it twice for the first one no?

Reply:
hum I have added that because I had a bug on these line, you want me to remove it ?